### PR TITLE
Added tcpdump file fetch functionality

### PR DESCRIPTION
**Pull request by danielmgit**
furthermore refactored. Also compressed 007-kill.

netperf applet: background running functionality added, also refactored.

001-netperf-data-exchange/run.cmd adjusted to current status
005-netperf was successfully tested locally
006-tcpdump was successfully tested locally
008-wait-for-completion was successfully tested locally

### CODE REVIEW

File: applets/006-tcpdump/applet.py [hunk 1]
```diff
@@ -1,91 +1,140 @@
 
-def main(x, conf, args):
+import os
+import subprocess
+
+
+def tcpdump_start(x, arg_d):
+    _, _, _ = x.ssh.exec(arg_d["host_ip"], arg_d["host_user"],
```
Comment (hgn) [inline applets/006-tcpdump/applet.py]: For the case all return values are ignored no need to catch the values at all:

```
 x.ssh.exec(arg_d["host_ip"], ...)
```

File: applets/006-tcpdump/applet.py [hunk 2]
```diff
@@ -1,91 +1,140 @@
...
+def transfer_dumpfile(x, arg_d):
+
+    # make local directories
+    os.makedirs(arg_d["local_file_name"])
+    process = subprocess.Popen("scp {}@{}:/tmp/net-applet-shuffler/dump_{}.pcap"
+            " {}".format(arg_d["host_user"], arg_d["host_ip"],
+            arg_d["applet_id"], arg_d["local_file_name"]).split(),
+            stdout=subprocess.PIPE)
+    # block until file transfer is done
+    _, _ = process.communicate()
```
Comment (hgn) [inline applets/006-tcpdump/applet.py]: Same same

File: applets/006-tcpdump/applet.py [hunk 3]
```diff
@@ -1,91 +1,140 @@
...
+                         arg_d["filter"]) in x:
+            pid_tcpdump = x.split()[1]
+
+    # save pid to file
+    x.ssh.exec(arg_d["host_ip"], arg_d["host_user"], "touch "
+            "/tmp/net-applet-shuffler/tcpdump_{}".format(arg_d["applet_id"]))
+    x.ssh.exec(arg_d["host_ip"], arg_d["host_user"], "sh -c \"echo '{}' > "
+            "/tmp/net-applet-shuffler/tcpdump_{}\"".format(pid_tcpdump,
+                                                           arg_d["applet_id"]))
+
```
Comment (hgn) [inline applets/006-tcpdump/applet.py:48]: Better:

```
   cmd = "/tmp/net-applet-shuffler/tcpdump_{}"
    x.ssh.exec(arg_d["host_ip"], arg_d["host_user"], "touch " cmd.format(arg_d["applet_id"]))
    x.ssh.exec(arg_d["host_ip"], arg_d["host_user"], "sh -c \"echo '{}' > ""cmd\"".format(pid_tcpdump,
                      arg_d["applet_id"]))
```

File: applets/006-tcpdump/applet.py [hunk 4]
```diff
@@ -1,91 +1,140 @@
...
+    return True
+
+
+def tcpdump_stop(x, arg_d):
+    # retrieve pid
+    stdout, _, exit_code = x.ssh.exec(arg_d["host_ip"], arg_d["host_user"],
+            "echo \"$(</tmp/net-applet-shuffler/tcpdump_{})\""
+            .format(arg_d["applet_id"]))
+    pid_tcpdump = stdout.decode("utf-8").splitlines()[0]
```
Comment (hgn) [inline applets/006-tcpdump/applet.py]: exit code check should done before stdout.decode

File: applets/006-tcpdump/applet.py (hunk 1 above)
Comment (danielmgit) [inline applets/006-tcpdump/applet.py]: Thy shall be done!

File: applets/006-tcpdump/applet.py (hunk 3 above)
Comment (danielmgit) [inline applets/006-tcpdump/applet.py:48]: Indeed.

File: applets/006-tcpdump/applet.py (hunk 4 above)
Comment (danielmgit) [inline applets/006-tcpdump/applet.py]: Check!

**Comment by hgn** [comment]
Looks great - applied!

**Pull request merged**